src/Common/REST/TEC/V1/Traits/Read_Custom_Archive_Response.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * Trait to handle the response for read custom archive requests.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\REST\TEC\V1\Traits
+ */
+
+declare( strict_types=1 );
+
+namespace TEC\Common\REST\TEC\V1\Traits;
+
+use WP_REST_Response;
+use TEC\Common\Contracts\Repository_Interface;
+
+/**
+ * Trait to handle the response for read custom archive requests.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\REST\TEC\V1\Traits
+ */
+trait Read_Custom_Archive_Response {
+	/**
+	 * Handles the read request for the endpoint.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $params The sanitized parameters to use for the request.
+	 *
+	 * @return WP_REST_Response The response object.
+	 */
+	public function read( array $params = [] ): WP_REST_Response {
+		$page     = absint( $params['page'] ?? 1 );
+		$per_page = absint( $params['per_page'] ?? $this->get_default_posts_per_page() );
+
+		/** @var Repository_Interface $query */
+		$query = $this->build_query( $params );
+
+		$query->page( $page )->per_page( $per_page );
+
+		$data  = $this->format_entity_collection( $query->all() );
+		$total = $query->found();
+
+		// Return 404 if no entities found and page > 1.
+		if ( empty( $data ) && $page > 1 ) {
+			return new WP_REST_Response(
+				[
+					'code'    => 'tec_rest_organizers_page_not_found',
+					'message' => __( 'The requested page was not found.', 'tribe-common' ),
+				],
+				404
+			);
+		}
+
+		/**
+		 * Filters the data that will be returned for an entities archive request.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $data   The retrieved data.
+		 * @param array $params The sanitized parameters to use for the request.
+		 */
+		$data = apply_filters( 'tec_rest_' . $this->get_model_class() . '_archive', $data, $params );
```

Comment:
2 things:

1. Add `get_model_class()` as an abstract.
2. String interpolation is easier to read, IMO

```suggestion
		$data = apply_filters( "tec_rest_{$this->get_model_class()}_archive", $data, $params );
```